pkg/algorithm/nodeselect/v1alpha2/select_node.go

```diff
@@ -40,13 +40,20 @@ func (ac *Config) SelectNode() (*apis.PoolSpec, string, error) {
 		// pin it
 		pool := pool
 		nodeName, err := GetNodeFromLabelSelector(pool.NodeSelector)
-		if err != nil {
+		if err != nil || nodeName == "" {
```

Comment:
nit: IMO `nodeName == ""` is not required

Reply:
Just added an extra safe check after having discussion with @vishnuitta

Reply:
added for sake of safety in future for any code change in GetNodeFromLabelSelector.. code maintainability